docs/bundles/media-bundle/cropping-media-objects.md

```diff
@@ -0,0 +1,189 @@
+# Cropping media objects in the bundles
+
+## cropping
+
+Oftentimes before this implementation we needed to make pageparts with multiple media object uploads to be able to scale up or down for desktops to mobile phones.
+Now we have made a feature that allows you to use one media object and crop it in several different ways for each viewport.
+
+## Implementation
+
+To use a croppable media object you want to use the wrapper object instead of the actual media object. If you do that all the rest will be done automatically.
+
+pagepart.php
+```php
+<?php
+
+namespace App\Entity\PageParts;
+
+use App\Form\PageParts\CroppableImagePagePartAdminType;
+use Doctrine\ORM\Mapping as ORM;
+use Kunstmaan\MediaBundle\Entity\EditableMediaWrapper;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="app_croppable_image_page_parts")
+ */
+class CroppableImagePartPart extends AbstractPagePart
+{
+    /**
+     * @ORM\ManyToOne(targetEntity="Kunstmaan\MediaBundle\Entity\EditableMediaWrapper", cascade={"persist"})
+     * @ORM\JoinColumn(name="media_id", referencedColumnName="id")
+     * @Assert\NotNull()
```

Comment:
I think this should be a `@Assert\Valid()` as the NotNull is enforced on the Wrapper entity media property.

This and the other comments below  👇 should be checked/applied to the classes in the generator bundle